### ISSUE TITLE
query in Energy Based GAN (EBGAN)

### ISSUE THREAD

**Issue by devraj89** [eriklindernoren/PyTorch-GAN#18]
Hi 
Thank you for your wonderful effort in implementing so many papers.
I have a query regarding your EBGAN implementation.
https://github.com/eriklindernoren/PyTorch-GAN/blob/master/implementations/ebgan/ebgan.py

In `line 175` when you are optimizing the generator `G` why is the `pixelwise_loss` computed using `gen_imgs.detach()` and why not simply `gen_imgs()` ? If we do `.detach()` while updating `G` then the `pixelwise_loss` will not contribute any gradients towards optimizing the generator weights. Is it the right way to do that ?

Please clarify my doubt.
Thank You in Advance !

**Comment by eriklindernoren** [comment]
Because we use `gen_imgs` as the target variable in that case. The generator is updated according to the `recon_imgs` output.

**Comment by devraj89** [comment]
Thank you for your prompt reply ! It is an excellent catch. But does that mean that the Generator outputs `gen_imgs` are not directly being used to drive any weight updates but rather indirectly used when we use the reconstructions of the generated outputs i.e. `recon_imgs`?

**Comment by eriklindernoren** [comment]
Sorry for the late response. Yes, your observation is correct. :)
